app/java-2/src/main/java/com/google/ProcessingJob.java

```diff
@@ -0,0 +1,142 @@
+        Dictionary<String, Integer> colorAgeLocationToCount = new Hashtable<String, Integer>();
+        int numOfProcessedRows = 0;
+        int numOfIgnoredRows = 0;
+        // Go through each row of the CSV file.
+        FileReader reader = new FileReader(tempDataFilePath);
+        CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader());
+        for (CSVRecord record : csvParser) {
+            numOfProcessedRows++;
+            // Ignore rows where crucial columns are empty/unknown.
+            boolean shouldIgnoreRow = false;
+            for (String column : CSV_COLUMNS) {
+                String value = record.get(column);
+                if (value == null || value.isEmpty() || value.equals("?")) {
+                    numOfIgnoredRows++;        
+                    shouldIgnoreRow = true;
+                }
+            }
+
+            // Bump count in "Color/Age/Location" dictionary.
+            if (!shouldIgnoreRow) {
+                String key = String.format(
+                    "%s/%s/%s",
+                    record.get("Primary Fur Color"),
+                    record.get("Age"),
+                    record.get("Location")
+                );
+                Integer currCount = colorAgeLocationToCount.get(key);
+                if (currCount == null) {
+                    currCount = 0;
+                }
+                colorAgeLocationToCount.put(key, currCount + 1);
+            }
+        }
+        System.out.println(String.format(
+            "processRawData: processed %s records, removed %s",
+            numOfProcessedRows, 
+            numOfIgnoredRows));
+        return colorAgeLocationToCount;
+    }
+
+    /**
+     * Uploads a JSON file (to a Google Cloud Storage bucket)
+     * for each combination of facets (for each color-age-location combination).
+     */
+    public static void writeProcessedData(Dictionary<String, Integer> colorAgeLocationToCount) 
+        throws UnsupportedEncodingException {
+        System.out.println("writeProcessedData: start writing");
+        Gson gson = new Gson();
+        Enumeration<String> keys = colorAgeLocationToCount.keys();
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            String jsonString = gson.toJson(colorAgeLocationToCount.get(key));
+            BlobId cloudStorageBlobId = BlobId.of(PROCESSED_DATA_BUCKET, key + "/data.json");
+            BlobInfo cloudStorageBlobInfo = BlobInfo
+                .newBuilder(cloudStorageBlobId)
+                .setContentType("application/json")
+                .build();
+            String utf8CharsetName = StandardCharsets.UTF_8.name();
+            byte[] jsonStringAsByes = jsonString.getBytes(utf8CharsetName);
+            storage.create(cloudStorageBlobInfo, jsonStringAsByes, 0, jsonStringAsByes.length);
```

Comment:
This `storage.create(...)` method is [defined and documented here](https://github.com/googleapis/java-storage/blob/3a0b82929f2c26e2f26f30266319e25bdbe1c5a7/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java#L2830).